### DISCUSSION TITLE
Are ICMP requests possible?

### DISCUSSION THREAD

**Discussion by JaneX8** [projectdiscovery/nuclei#2938] [Q&A] [answered]
Is it possible to do ICMP requests? I'd like to start by simply performing a regular "ping", an ICMP echo request see if there is any reply.

**Comment by forgedhallpass** [comment, marked as answer]
1. [Host discovery](https://github.com/projectdiscovery/naabu/issues/332) is already supported in `naabu`, so you could do your scan and then simply pipe the results to `nuclei`.

2. There's also an existing ticket to [integrate `naabu` into `nuclei`](https://github.com/projectdiscovery/nuclei/issues/2558). Once implemented you could potentially control these requests from templates as well.

Are these what you were looking for?

**Comment by bandrel** [comment]
Understanding that host discovery is out of scope for nuclei, ICMP would be necessary to check things like if ICMP type 13  responds with ICMP type 14. This function would be a vulnerability check and not a host discovery. Example CVE-1999-0524.

Is there a way to leverage naabu to accomplish this task?
